packages/@tests/marine/web/test/test.spec.ts

```diff
@@ -0,0 +1,86 @@
+import Webpack from 'webpack';
+import WebpackDevServer from 'webpack-dev-server';
+import webpackConfig from '../webpack.config.js';
+import process from 'process';
+import path from 'path';
+import fs from 'fs';
+
+// change directory to the location to the test-project.
+// run all the subsequent Webpack scripts in that directory
+process.chdir(path.join(__dirname, '..'));
+
+let server;
+
+jest.setTimeout(10000);
+
+const startServer = async (modifyConfig?) => {
+    const loadInBrowserToDebug = false;
+    // const loadInBrowserToDebug = true; // use this line to debug
+
+    modifyConfig = modifyConfig || ((_) => {});
+
+    const cfg: any = webpackConfig();
+    modifyConfig(cfg);
+    const compiler = Webpack(cfg);
+    const devServerOptions = { ...cfg.devServer, open: loadInBrowserToDebug };
+    server = new WebpackDevServer(devServerOptions, compiler);
+    await server.start();
+    // wait for webpack to load
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+};
+
+const stopServer = async () => {
+    console.log('test: stopping server');
+    await server.stop();
+};
+
+const publicDir = 'public';
+
+function copyFile(packageName: string, fileName: string) {
+    const modulePath = require.resolve(packageName);
+    const source = path.join(path.dirname(modulePath), fileName);
+    const dest = path.join(publicDir, fileName);
+
+    fs.copyFileSync(source, dest);
+}
+
+const copyPublicDeps = async () => {
+    fs.mkdirSync(publicDir, { recursive: true });
+    copyFile('@fluencelabs/marine.worker-script', 'marine-js.web.js');
+    copyFile('@fluencelabs/marine-js', 'marine-js.wasm');
+    copyFile('@fluencelabs/avm', 'avm.wasm');
+};
+
+const cleanPublicDeps = async () => {
+    fs.rmSync(publicDir, { recursive: true, force: true });
+};
+
+describe('Browser integration tests', () => {
+    beforeEach(async () => {
+        await copyPublicDeps();
```

Comment:
don't need await if everything is sync above. But if it's possible I would rather suggest to make everything async and copy files with `await Promise.all`. It's a super simple one minute refactor

Reply:
fixed